sdks/java/io/hcatalog/src/main/java/org/apache/beam/sdk/io/hcatalog/HCatalogIO.java

```diff
@@ -196,11 +237,27 @@ Read withContext(ReaderContext context) {
     }
 
     @Override
+    @SuppressWarnings("deprecation")
     public PCollection<HCatRecord> expand(PBegin input) {
       checkArgument(getTable() != null, "withTable() is required");
       checkArgument(getConfigProperties() != null, "withConfigProperties() is required");
-
-      return input.apply(org.apache.beam.sdk.io.Read.from(new BoundedHCatalogSource(this)));
+      Watch.Growth<Read, Integer, Integer> growthFn;
+      if (getPollingInterval() != null) {
+        growthFn = Watch.growthOf(new PartitionPollerFn()).withPollInterval(getPollingInterval());
+        if (getTerminationCondition() != null) {
+          growthFn =
```

Comment:
Sorry I was maybe not clear, I am talking about not repeating agan the code `Watch.growthOf(new PartitionPollerFn()).withPollInterval(getPollingInterval())` because that's already done just before in the definition of `growthFn` and instead do: `growthFn = growthFn.withTerminationPerInput(getTerminationCondition());`.

It is pretty minor but nice to fix.

Reply:
Ah. I see. Done